pysqldb3/geopackage.py

```diff
@@ -290,79 +281,88 @@ def gpkg_to_shp(self, print_cmd=False):
                 print(ogr_response)
             except subprocess.CalledProcessError as e:
                 print("Ogr2ogr Output:\n", e.output)
                 print('Ogr2ogr command failed. The Geopackage/feature class was not written.')
-                raise subprocess.CalledProcessError(cmd=print_cmd_string([self.dbo.password], self.cmd), returncode=1)
         
         return
 
-    def table_exists(self):
+    def table_exists(self, dbo = None, schema = None, table = None):
         """
         Wrapper for DbConnect table_exists function.
 
+        :param table: table name
+        :param schema: schema name
         :return: boolean if gpkg name exists as a table in the schema
         """
 
-        return self.dbo.table_exists(table=self.table, schema=self.schema)
+        return dbo.table_exists(schema = schema, table = table)
 
-    def del_indexes(self):
+    def del_indexes(self, dbo, schema = None, table = None):
         """
         Drops indexes
+        :param dbo: database connection
+        :param schema: Schema for deleting index
+        :param table: Table for deleting index
         :return:
         """
-        if self.dbo.type == 'PG':
-            self.dbo.query(SHP_DEL_INDICES_QUERY_PG.format(s=self.schema, t=self.table), internal=True)
-            indexes_to_delete = self.dbo.internal_data
+        
+        if dbo.type == 'PG':
+            dbo.query(SHP_DEL_INDICES_QUERY_PG.format(s=schema, t=table), internal=True)
+            indexes_to_delete = dbo.internal_data
 
             for _ in list(indexes_to_delete):
                 table_name, schema_name, index_name, column_name = _
                 if 'pkey' not in index_name and 'PK' not in index_name:
-                    self.dbo.query('DROP INDEX {s}.{i}'.format(s=self.schema, i=index_name),
+                    dbo.query('DROP INDEX {s}.{i}'.format(s=schema, i=index_name),
                                    strict=False, internal=True)
         else:
-            self.dbo.query(SHP_DEL_INDICES_QUERY_MS.format(s=self.schema, t=self.table), internal=True)
-            indexes_to_delete = self.dbo.internal_data
+            dbo.query(SHP_DEL_INDICES_QUERY_MS.format(s=schema, t=table), internal=True)
+            indexes_to_delete = dbo.internal_data
 
             for _ in list(indexes_to_delete):
                 table_name, index_name, column_name, idx_typ = _
                 if 'pkey' not in index_name and 'PK' not in index_name:
-                    self.dbo.query('DROP INDEX {t}.{i}'.format(t=self.table, i=index_name), strict=False, internal=True)
+                    dbo.query('DROP INDEX {t}.{i}'.format(t=table, i=index_name), strict=False, internal=True)
 
-    def read_gpkg(self, precision=False, private=True, gpkg_encoding=None, print_cmd=False):
+    def read_gpkg(self, dbo = None, schema = None, table = None, port = 5432, srid = '2263', gdal_data_loc=GDAL_DATA_LOC, precision=False, private=True, gpkg_encoding=None, print_cmd=False):
```

Comment:
if the output is always a database table the dbo should be required. Also i think  private should default to False